<launch>

  <!-- <arg name="stdout" default="screen" /> -->
  <arg name="cloud_topic" default="laser_points" />
  <arg name="sensor_id" default="0" />

  <!-- <node pkg="nodelet" exec="nodelet" name="pcl_manager_$(var sensor_id)" args="manager" output="screen"/>

  <node pkg="nodelet" exec="nodelet" name="pass_through_z_velo_$(var sensor_id)" args="load pcl/PassThrough pcl_manager_$(var sensor_id)" output="screen">
    <remap from="~input" to="$(var cloud_topic)" />
    <remap from="~output" to="/lidar_pattern_$(var sensor_id)/z_filtered" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -2
      filter_limit_max: 1
      filter_limit_negative: False
      max_queue_size: 1
    </rosparam>
  </node>

  <node pkg="nodelet" exec="nodelet" name="pass_through_y_velo_$(var sensor_id)" args="load pcl/PassThrough pcl_manager_$(var sensor_id)" output="screen">
    <remap from="~input" to="/lidar_pattern_$(var sensor_id)/z_filtered" />
    <remap from="~output" to="/lidar_pattern_$(var sensor_id)/zy_filtered" />
    <rosparam>
      filter_field_name: y
      filter_limit_min: -100
      filter_limit_max: 100
      filter_limit_negative: False
      max_queue_size: 1
    </rosparam>
  </node>

  <node pkg="nodelet" exec="nodelet" name="pass_through_x_velo_$(var sensor_id)" args="load pcl/PassThrough pcl_manager_$(var sensor_id)" output="screen">
    <remap from="~input" to="/lidar_pattern_$(var sensor_id)/zy_filtered" />
    <remap from="~output" to="/lidar_pattern_$(var sensor_id)/zyx_filtered" />
    <rosparam>
      filter_field_name: x
      filter_limit_min: 0
      filter_limit_max: 100
      filter_limit_negative: False
      max_queue_size: 1
    </rosparam>
  </node> -->

  <!-- <node pkg="calibration" exec="lidar_pattern" name="lidar_pattern_$(var sensor_id)" output="screen">
    <remap from="~cloud1" to="/lidar_pattern_$(var sensor_id)/zyx_filtered"/>
    <rosparam>
      passthrough_radius_min: 1.0
      passthrough_radius_max: 6.0
    </rosparam>
  </node> -->

  <node pkg="calibration" exec="lidar_pattern" name="lidar_pattern_$(var sensor_id)" output="screen">
    <remap from="cloud1" to="$(var cloud_topic)"/>
    <remap from="centers_cloud" to="/lidar_pattern_$(var sensor_id)/centers_cloud"/>
    <param name="passthrough_radius_min" value="1.0" />
    <param name="passthrough_radius_max" value="6.0" />
  </node>

</launch>
